agent/engine/docker_task_engine.go

```diff
@@ -616,41 +625,120 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 	containerMap, ok := engine.state.ContainerMapByArn(task.Arn)
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container belongs to unrecognized task %s", task.Arn)},
+			Error: CannotStartContainerError{errors.Errorf("Container belongs to unrecognized task %s", task.Arn)},
 		}
 	}
 
 	dockerContainer, ok := containerMap[container.Name]
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
+			Error: CannotStartContainerError{errors.Errorf("Container not recorded as created")},
 		}
 	}
 	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
 }
 
 func (engine *DockerTaskEngine) provisionContainerResources(task *api.Task, container *api.Container) DockerContainerMetadata {
-	seelog.Infof("Task [%v]: Setting up container resources for container [%v]", task.String(), container.String())
+	seelog.Infof("Task [%s]: Setting up container resources for container [%s]", task.String(), container.String())
+	cniConfig, err := engine.buildCNIConfigFromTaskContainer(task, container)
+	if err != nil {
+		return DockerContainerMetadata{
+			DockerID: cniConfig.ContainerID,
+			Error:    ContainerNetworkingError{errors.Wrap(err, "provisionContainerResources: build cni configuration error")},
```

Comment:
Can you replace the string `"provisionContainerResources"` here and in all other error messages to be more user friendly? Something like `"container resources provisioning: unable to build cni configuration"`